### PR TITLE
Backport: revert non-ajax support for select2 (analog to [d46d150])

### PR DESCRIPTION
**Pull request by tgoetz**

### CODE REVIEW

File: select2-parent/select2/src/main/java/org/wicketstuff/select2/AbstractSelect2Choice.java [hunk 1]
```diff
@@ -115,66 +110,6 @@ public AbstractSelect2Choice(String id, IModel<M> model, ChoiceProvider<T> provi
...
-	 * Construct.
-	 *
-	 * @param id
-	 *            markup id
-	 * @param model
-	 *            model for select
-	 * @param collection
-	 *            list options for select
-	 */
-	public AbstractSelect2Choice(String id, IModel<M> model, List<T> collection)
```
Comment (solomax) [inline select2-parent/select2/src/main/java/org/wicketstuff/select2/AbstractSelect2Choice.java]: Why there should be no possibility to create Select2Choice with "pre-selected" options?

File: select2-parent/select2-examples/src/main/java/org/wicketstuff/select2/HomePage.java [hunk 2]
```diff
@@ -12,20 +12,16 @@
...
-import org.apache.wicket.markup.html.form.IChoiceRenderer;
-import org.apache.wicket.model.IModel;
-import org.apache.wicket.model.Model;
 import org.apache.wicket.model.PropertyModel;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
```
Comment (solomax) [inline select2-parent/select2-examples/src/main/java/org/wicketstuff/select2/HomePage.java]: Not sure why this block of imports should change the position

File: select2-parent/select2-examples/src/main/java/org/wicketstuff/select2/HomePage.java [hunk 3]
```diff
@@ -70,19 +66,6 @@ public HomePage()
...
-		Form<Void> clientSideSelect2Form = new Form<Void>("single");
-		clientSideSelect2ChoiceWrapper.add(clientSideSelect2Form);
-		Select2Choice<Country> clientSideSelect2Choice = new Select2Choice<Country>("country",
-				(IModel<Country>) clientSideSelect2ChoiceWrapper.getDefaultModel(), Arrays.asList(Country.values()),
-				new CountryChoiceRenderer());
-		clientSideSelect2Choice.getSettings().setAllowClear(true);
-		clientSideSelect2Choice.getSettings().setPlaceholder("Select a country from the list...");
-		clientSideSelect2Form.add(clientSideSelect2Choice);
-		add(clientSideSelect2ChoiceWrapper);
 	}
```
Comment (solomax) [inline select2-parent/select2-examples/src/main/java/org/wicketstuff/select2/HomePage.java]: I believe there should be at least one example for "single select"

File: select2-parent/select2/src/main/java/org/wicketstuff/select2/AbstractSelect2Choice.java [hunk 4]
```diff
@@ -28,14 +21,18 @@
...
 import org.apache.wicket.request.http.WebRequest;
 import org.apache.wicket.request.http.WebResponse;
 import org.apache.wicket.util.string.Strings;
-import org.wicketstuff.select2.json.JsonBuilder;
+
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
```
Comment (solomax) [inline select2-parent/select2/src/main/java/org/wicketstuff/select2/AbstractSelect2Choice.java]: same comment on import reordering

**Comment by solomax** [comment]
Additionally you have dropped lots of tests ....

File: select2-parent/select2/src/main/java/org/wicketstuff/select2/AbstractSelect2Choice.java (hunk 1 above)
Comment (tgoetz) [inline select2-parent/select2/src/main/java/org/wicketstuff/select2/AbstractSelect2Choice.java]: First of all: you don't need a Wicket component for that, just create your (static) dropdown and use select2 directly. We could consider implementing a Select2Behavior for that.

Second: the way this is implemented forces the user to supply a static List of choices, else an exception is thrown during construction.

This was already discussed and accepted for master branch, just backporting it now as I'm currently using wicket-6.x in my $ project.

File: select2-parent/select2-examples/src/main/java/org/wicketstuff/select2/HomePage.java (hunk 2 above)
Comment (tgoetz) [inline select2-parent/select2-examples/src/main/java/org/wicketstuff/select2/HomePage.java]: IntelliJ code formating, pls ignore ;)
Can also undo that, but hey, it's just imports ...

File: select2-parent/select2-examples/src/main/java/org/wicketstuff/select2/HomePage.java (hunk 3 above)
Comment (tgoetz) [inline select2-parent/select2-examples/src/main/java/org/wicketstuff/select2/HomePage.java]: There is! Just removed the example with the 'static' list.

**Comment by tgoetz** [comment]
No, only those test that tested the (newly introduced) non-ajax functionality with static lists.

**Comment by tgoetz** [comment]
Please see related commit in master (d46d150) ...

File: select2-parent/select2/src/main/java/org/wicketstuff/select2/AbstractSelect2Choice.java (hunk 1 above)
Comment (solomax) [inline select2-parent/select2/src/main/java/org/wicketstuff/select2/AbstractSelect2Choice.java]: Sorry, my bad
missed this is the fix for 6.x branch, will merge it right now

**Pull request merged**